src/agents/common/state.py

```diff
@@ -0,0 +1,65 @@
+import operator
+from collections.abc import Sequence
+from enum import Enum
+from typing import Annotated
+
+from langchain_core.messages import BaseMessage
+from langchain_core.pydantic_v1 import BaseModel, Field
+
+
+class SubTaskStatus(str, Enum):
+    """Status of the sub-task."""
+
+    PENDING = "pending"
+    COMPLETED = "completed"
+
+
+class SubTask(BaseModel):
+    """Sub-task data model."""
+
+    description: str = Field(description="description of the task")
+    assigned_to: str = Field(description="agent to whom the task is assigned")
+    status: str = Field(default=SubTaskStatus.PENDING)
+    result: str | None
+
+    def complete(self) -> None:
+        """Update the result of the task."""
+        self.status = SubTaskStatus.COMPLETED
+
+
+# After upgrading generative-ai-hub-sdk we can message that use pydantic v2
+# Currently, we are using pydantic v1.
+class UserInput(BaseModel):
```

Comment:
This is duplicate of `agents.common.data.Message`. Why we need it?

Reply:
yes, there is langchain problem with Pydantic. Generative ai hub sdk needs little older langchain which only work with pydantic v1. This blocked to use pydantic v2. I had to duplicate at the moment. As soon as we upgrade generative ai hub sdk and langchain that supports latest pydantic versions. We can directly use Message class using pydantic v2.
FastAPI with Depends didn't work with pydantic v1. Hence, I was in deadlock and introduced this workaround. 

Let's upgrade generative ai hub sdk in a next step as they introduce huge changes. I haven't trieid it yet.